code/lib/router/src/utils.test.ts

```diff
@@ -137,12 +137,12 @@ describe('buildArgsParam', () => {
 
   it('builds arrays', () => {
     const param = buildArgsParam({}, { arr: ['1', '2', '3'] });
-    expect(param).toEqual('arr[0]:1;arr[1]:2;arr[2]:3');
+    expect(param).toEqual('arr%5B0%5D:1;arr%5B1%5D:2;arr%5B2%5D:3');
```

Comment:
Let's not do that. This would be a breaking change and it looks extremely ugly. I know it's not standards compliant to keep unencoded things in the URL, but it works fine in all browsers and these are only parsed on the client-side (no server involved).